### PR TITLE
fix: tx fields for accounts allowing empty string

### PR DESCRIPTION
**Pull request by ckniffen**
## High Level Overview of Change

Fields that were encoded as STAccounts were turning `''` into `rrrrrrrrrrrrrrrrrrrrrhoLvTp` the ACCOUNT_ZERO account. They will now throw a `ValidationError`.

Invalid addresses on a transaction now throws a `ValidationError` when submitting a transaction instead of `Error('checksum_invalid')`.

This change updates some error messages for fields a few fields that aren't accounts, `DestinationTag` or `NFTokenID`.  They will still be of type `ValidationError`.

Fixes: #2517 and #2475

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### Did you update HISTORY.md?

- [x] Yes
- [ ] No, this change does not impact library users

### CODE REVIEW

File: packages/xrpl/test/models/NFTokenCreateOffer.test.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ describe('NFTokenCreateOffer', function () {
       TransactionType: 'NFTokenCreateOffer',
       NFTokenID: NFTOKEN_ID,
       Amount: '1',
-      Owner: 'r9LqNeG6qHxjeUocjvVki2XR35weJ9mZgQ',
+      Owner: 'rcXY84C4g14iFp6taFXjjQGVeHqSCh9RX',
```
Comment (mvadari) [inline packages/xrpl/test/models/NFTokenCreateOffer.test.ts:19]: Why were all of these changed?

File: packages/xrpl/test/models/payment.test.ts [hunk 2]
```diff
@@ -116,12 +116,26 @@ describe('Payment', function () {
...
     )
     assert.throws(
       () => validate(paymentTransaction),
       ValidationError,
-      'PaymentTransaction: invalid Destination',
+      'Payment: invalid field Destination',
+    )
+  })
+
+  it(`throws when Destination is invalid classic address`, function () {
```
[the remark this answers is not in the document]
Comment (mvadari) [reply packages/xrpl/test/models/payment.test.ts:128]: Can there also be a test for an empty string?

File: packages/xrpl/test/models/NFTokenCreateOffer.test.ts (hunk 1 above)
Comment (ckniffen) [reply packages/xrpl/test/models/NFTokenCreateOffer.test.ts:19]: The previous value was an invalid classic address.  Looking at it now I could have just deleted a character.

Comment (ckniffen) [reply packages/xrpl/test/models/NFTokenCreateOffer.test.ts:19]: That didnt work because it is still not valid.  The examples lower were too short.

File: packages/xrpl/test/models/NFTokenCreateOffer.test.ts [hunk 3]
```diff
@@ -36,7 +36,7 @@ describe('NFTokenCreateOffer', function () {
         tfSellNFToken: true,
       },
       Expiration: 1000,
-      Destination: 'r9LqNeG6qHxjeUocjvVki2XR35weJ9mZgQ',
+      Destination: 'rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn',
```
Comment (ckniffen) [inline packages/xrpl/test/models/NFTokenCreateOffer.test.ts]: Reverting this one because it did not need to change.

File: packages/xrpl/HISTORY.md [hunk 4]
```diff
@@ -9,10 +9,12 @@ Subscribe to [the **xrpl-announce** mailing list](https://groups.google.com/g/xr
 * Add pseudo transaction types to `tx` and `ledger` method responses.
 * Add missing `type` param to `ledger_data` and `ledger` requests
 * Type assertions around `PreviousTxnID` and `PreviousTxnLgrSeq` missing on some ledger objects
+* Transaction fields that equate to an address no longer allow an empty string `''`. If you want to specify the genesis account you must manually specify `rrrrrrrrrrrrrrrrrrrrrhoLvTp`
```
Comment (intelliot) [inline packages/xrpl/HISTORY.md:12]:
```suggestion
* Transaction fields that represent an address no longer allow an empty string (`''`). If you want to specify [ACCOUNT_ZERO](https://xrpl.org/addresses.html#special-addresses), you can specify `rrrrrrrrrrrrrrrrrrrrrhoLvTp`. ⚠️ **WARNING:** `rrrrrrrrrrrrrrrrrrrrrhoLvTp` is a black hole address, with no corresponding private key. Accounts/funds controlled by this address are not accessible.
```

**Review by intelliot — CHANGES_REQUESTED** [review]
The genesis account is `rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh`, not `rrrrrrrrrrrrrrrrrrrrrhoLvTp`.

`rrrrrrrrrrrrrrrrrrrrrhoLvTp` is `0` (or the empty string) encoded with the XRPL's base58 address format. Named ACCOUNT_ZERO, it is a known black hole address, meaning that it was not derived from a secret key.

File: packages/xrpl/HISTORY.md (hunk 4 above)
Comment (mvadari) [reply packages/xrpl/HISTORY.md:12]: `rrrrrrrrrrrrrrrrrrrrrhoLvTp` is not the genesis account, it's the zero account

**Review by intelliot — APPROVED** [review]
